kong/templates/nginx_kong.lua

```diff
@@ -252,4 +252,37 @@ server {
     }
 }
 > end
+
+> if #status_listeners > 0 then
+server {
+    server_name kong_status;
+> for i = 1, #status_listeners do
+    listen $(status_listeners[i].listener);
+> end
+
+    access_log off;
```

Comment:
Wouldn't it be preferable to write access logs to the same location as the Admin API access logs? In any case, disabling them unconditionally seems a bit extreme and users may feel blind spotted.

Reply:
Given that this is a health-check and metrics endpoint in the current form, in most deployments, it is likely users will not want access logs for requests to this endpoint to be logged.
We can move this as an injected directive inside `kong.conf`.
It seems like a sane and an important default.
Thoughts?

Reply:
this seems to be the only missing point.  is this a blocker to merge?

Reply:
I'm not a fan of adding yet another `*_access_log` config property, especially given that it can easily be added with an NGINX dynamic directive (`nginx_http_status_access_log = logs/status_access.log`). However, I'm also not totally satisfied with having to let users figure this out by themselves.

Over the last few years, I have often seen users being particularly demanding about keeping access logs enabled for all exposed listeners for observability's sake.

**Side note**: as of today, errors in the Status API would be logged to the proxy server error logs since we do not override the `error_log` directive in the status `server {}` block. **I consider this to be a blocker as well**. 

My suggestion for now would be to follow the already established pattern and expose new `status_error_log` and `status_access_log` properties (which can be disabled by default, I think that's fine). This way users have a clear and easy way to enable logging on this new interface.

---

#### Side note

In my opinion, the friendliest approach we could take given the explosion of new API interfaces exposed by Kong (Community and Enterprise) would be to include the `$server_name` variable in the default `access_log` format used by Kong, and log all API accesses in the same file by default.

E.g.:

```
logs/
├── admin_access.log
├── admin_error.log
├── proxy_access.log
└── proxy_error.log
```

Where all APIs (Admin, Status, etc...) write to `admin_access.log` by default, but including the server name (`kong_admin`, `kong_status`, etc...) is included in each log line. Users can then decide to split specific interfaces to their own files or disable them altogether (or even configure fancier `access_log` directives if desired).

Reply:
Agreed.
Updated the code to include `status_access_log` and `status_error_log` configuration properties which default to `off` and `logs/status_error.log` respectively.